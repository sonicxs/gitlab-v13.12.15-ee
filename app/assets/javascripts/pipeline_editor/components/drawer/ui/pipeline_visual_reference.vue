<script>
import { s__ } from '~/locale';
import DemoJobPill from './demo_job_pill.vue';

export default {
  i18n: {
    stageNames: {
      build: s__('StageName|Build'),
      test: s__('StageName|Test'),
      deploy: s__('StageName|Deploy'),
    },
    jobNames: {
      build: s__('JobName|build-job'),
      test_1: s__('JobName|unit-test'),
      test_2: s__('JobName|lint-test'),
      deploy: s__('JobName|deploy-app'),
    },
  },
  stageClasses:
    'gl-bg-blue-50 gl-display-flex gl-flex-direction-column gl-align-items-center gl-p-4 gl-rounded-base',
  titleClasses: 'gl-text-blue-600 gl-mb-4',
  components: {
    DemoJobPill,
  },
};
</script>
<template>
  <div class="gl-display-flex gl-justify-content-center">
    <div :class="$options.stageClasses" class="gl-mr-5">
      <div :class="$options.titleClasses">{{ $options.i18n.stageNames.build }}</div>
      <demo-job-pill :job-name="$options.i18n.jobNames.build" />
    </div>
    <div :class="$options.stageClasses" class="gl-mr-5">
      <div :class="$options.titleClasses">{{ $options.i18n.stageNames.test }}</div>
      <demo-job-pill class="gl-mb-3" :job-name="$options.i18n.jobNames.test_1" />
      <demo-job-pill :job-name="$options.i18n.jobNames.test_2" />
    </div>
    <div :class="$options.stageClasses">
      <div :class="$options.titleClasses">{{ $options.i18n.stageNames.deploy }}</div>
      <demo-job-pill :job-name="$options.i18n.jobNames.deploy" />
    </div>
  </div>
</template>
