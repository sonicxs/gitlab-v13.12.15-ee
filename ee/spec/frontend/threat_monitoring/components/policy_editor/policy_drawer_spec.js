import { GlFormTextarea } from '@gitlab/ui';
import { shallowMount } from '@vue/test-utils';
import fromYaml from 'ee/threat_monitoring/components/policy_editor/lib/from_yaml';
import toYaml from 'ee/threat_monitoring/components/policy_editor/lib/to_yaml';
import PolicyDrawer from 'ee/threat_monitoring/components/policy_editor/policy_drawer.vue';
import PolicyPreview from 'ee/threat_monitoring/components/policy_editor/policy_preview.vue';

describe('PolicyDrawer component', () => {
  let wrapper;
  const policy = {
    name: 'test-policy',
    description: 'test description',
    endpointLabels: '',
    rules: [],
  };
  const unsupportedYaml = 'unsupportedPrimaryKey: test';

  const findPolicyPreview = () => wrapper.findComponent(PolicyPreview);
  const findTextForm = () => wrapper.findComponent(GlFormTextarea);

  const factory = ({ propsData } = {}) => {
    wrapper = shallowMount(PolicyDrawer, {
      propsData: {
        ...propsData,
      },
    });
  };

  afterEach(() => {
    wrapper.destroy();
  });

  describe('supported YAML', () => {
    beforeEach(() => {
      factory({ propsData: { value: toYaml(policy) } });
    });

    it('renders policy preview tabs', () => {
      expect(wrapper.find('div').element).toMatchSnapshot();
    });

    it('does render the policy description', () => {
      expect(findTextForm().exists()).toBe(true);
      expect(findTextForm().props()).toMatchObject({ value: 'test description' });
    });

    it('does render the policy preview', () => {
      expect(findPolicyPreview().exists()).toBe(true);
      expect(findPolicyPreview().props()).toStrictEqual({
        initialTab: 1,
        policyDescription: 'Deny all traffic',
        policyYaml: toYaml(policy),
      });
    });

    it('emits input event on description change', () => {
      wrapper.find(GlFormTextarea).vm.$emit('input', 'new description');

      expect(wrapper.emitted().input.length).toEqual(1);
      const updatedPolicy = fromYaml(wrapper.emitted().input[0][0]);
      expect(updatedPolicy.description).toEqual('new description');
    });
  });

  describe('unsupported YAML', () => {
    beforeEach(() => {
      factory({ propsData: { value: unsupportedYaml } });
    });

    it('renders policy preview tabs', () => {
      expect(wrapper.find('div').element).toMatchSnapshot();
    });

    it('does not render the policy description', () => {
      expect(findTextForm().exists()).toBe(false);
    });

    it('does render the policy preview', () => {
      expect(findPolicyPreview().exists()).toBe(true);
      expect(findPolicyPreview().props()).toStrictEqual({
        initialTab: 0,
        policyDescription: null,
        policyYaml: unsupportedYaml,
      });
    });
  });
});
