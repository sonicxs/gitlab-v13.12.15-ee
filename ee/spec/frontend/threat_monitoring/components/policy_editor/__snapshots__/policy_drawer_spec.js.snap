// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PolicyDrawer component supported YAML renders policy preview tabs 1`] = `
<div>
  <h4>
    Policy description
  </h4>
   
  <h5
    class="gl-mt-6"
  >
    Policy type
  </h5>
   
  <p>
    Network Policy
  </p>
   
  <div>
    <h5
      class="gl-mt-6"
    >
      Description
    </h5>
     
    <gl-form-textarea-stub
      noresize="true"
      value="test description"
    />
  </div>
   
  <policy-preview-stub
    class="gl-mt-4"
    initialtab="1"
    policydescription="Deny all traffic"
    policyyaml="apiVersion: cilium.io/v2
kind: CiliumNetworkPolicy
description: test description
metadata:
  name: test-policy
spec:
  endpointSelector:
    matchLabels:
      network-policy.gitlab.com/disabled_by: gitlab
"
  />
</div>
`;

exports[`PolicyDrawer component unsupported YAML renders policy preview tabs 1`] = `
<div>
  <h4>
    Policy description
  </h4>
   
  <h5
    class="gl-mt-6"
  >
    Policy type
  </h5>
   
  <p>
    Network Policy
  </p>
   
  <!---->
   
  <policy-preview-stub
    class="gl-mt-4"
    initialtab="0"
    policyyaml="unsupportedPrimaryKey: test"
  />
</div>
`;
